src/cases/services/search-filter.service.ts

```diff
@@ -1,20 +1,90 @@
 import { Injectable } from '@angular/core';
 import { SearchService } from '@hmcts/ccd-case-ui-toolkit';
 import { Observable } from 'rxjs';
+import { FormGroup } from '@angular/forms';
 
 @Injectable()
 export class SearchFilterService {
 
     constructor(
         private ccdSearchService: SearchService
-    ) {}
+    ) { }
+
+    metadataFields: string[];
 
     search(payload): Observable<any> {
-        const jurisdictionId = payload.selected.jurisdiction.id;
-        const caseTypeId = payload.selected.caseType.id;
-        const metaCriteria = payload.selected.metadataFields;
-        const caseCriteria = payload.selected.formGroup.value;
 
-        return this.ccdSearchService.search(jurisdictionId, caseTypeId, metaCriteria, caseCriteria) as any;
+        const filter = payload.selected;
+
+        let searchParams = {};
+
+        this.metadataFields = filter.metadataFields;
+
+        if (filter.caseState) {
+            searchParams = {
+                ...searchParams,
+                state: filter.caseState.id
+            };
+        }
+
+        if (filter.page) {
+            searchParams = {
+                ...searchParams,
+                page: filter.page
+            };
+        }
+
+        const jurisdictionId = filter.jurisdiction.id;
+        const caseTypeId = filter.caseType.id;
+        const filters = this.getCaseFilterFromFormGroup(filter.formGroup);
+        const caseFilters = filters.caseFilter;
+        const metadataFilters = Object.assign(searchParams, filters.metadataFilter);
+
+        return this.ccdSearchService.search(jurisdictionId, caseTypeId, metadataFilters, caseFilters) as any;
+    }
+
+    private getCaseFilterFromFormGroup(formGroup?: FormGroup): { caseFilter, metadataFilter } {
+        const result = {
+            caseFilter: {},
+            metadataFilter: {}
+        };
+
+        if (formGroup) {
+            this.buildFormDetails('', result, formGroup.value);
+        }
+        return result;
+    }
+
+    private buildFormDetails(parentPrefix: string, target: object, formGroupValue: object): void {
+        let prefix = parentPrefix;
+        if (parentPrefix && parentPrefix.length > 0) {
+            prefix = parentPrefix + '.';
+        }
+        for (let attributeName of Object.keys(formGroupValue)) {
+            const value = formGroupValue[attributeName];
+            if (this.isStringOrNumber(value)) {
+                const filterType = this.getFilterType(attributeName);
+                attributeName = this.sanitiseMetadataFieldName(filterType, attributeName);
+                target[filterType][prefix + attributeName] = value;
+            } else if (value) {
+                this.buildFormDetails(attributeName, target, value);
+            }
+        }
+    }
+
+    private isStringOrNumber(value: any): boolean {
```

Comment:
Probably this could be moved to a common Util.ts

Reply:
Just done this. Please re-review.